bootstrap/kubeadm/api/v1alpha3/conversion.go

```diff
@@ -49,13 +71,34 @@ func (dst *KubeadmConfigList) ConvertFrom(srcRaw conversion.Hub) error {
 // ConvertTo converts this KubeadmConfigTemplate to the Hub version (v1alpha4).
 func (src *KubeadmConfigTemplate) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*kubeadmbootstrapv1alpha4.KubeadmConfigTemplate)
-	return Convert_v1alpha3_KubeadmConfigTemplate_To_v1alpha4_KubeadmConfigTemplate(src, dst, nil)
+	if err := Convert_v1alpha3_KubeadmConfigTemplate_To_v1alpha4_KubeadmConfigTemplate(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually restore data.
+	restored := &kubeadmbootstrapv1alpha4.KubeadmConfigTemplate{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
```

Comment:
If `!ok` this error will return nil?

Reply:
True. I guess I should return error explicitly then. Other code use the same pattern though, which should also be fixed then.

Reply:
I see, I have no much context on these conversions. I'd propose to introduce the fixed pattern here and create a different issue for the existing ones. Unless there's a particular reason to continue with this pattern I'm not aware of cc @vincepri @CecileRobertMichon @fabriziopandini

Reply:
If the annotation doesn't exist, it should be a no-op. The logic makes sense to me as is.

Reply:
@enxebre do you mind if I mark it as resolved?